### ISSUE TITLE
self incorrectly marked as unsubscriptable in list subclass

### ISSUE THREAD

**Issue by pauladams8** [PyCQA/pylint#3930]
### Steps to reproduce
1. Subclass the builtin `list` class
2. Slice `self` from within the class

#### Example
```python
class CustomList(list):
    def first(self):
        # for example
        return self[0]
```

### Current behavior
pylint produces error
> Value 'self' is unsubscriptable

### Expected behavior
No alert from pylint

### pylint --version output
pylint 2.6.0
astroid 2.4.2
Python 3.8.5 (default, Jul 21 2020, 10:48:26) 
[Clang 11.0.3 (clang-1103.0.32.62)]

**Comment by hippo91** [comment]
@pauladams8 thanks for your report.
I'm unable to reproduce it.
Linting the following code:
```
#pylint:disable=missing-module-docstring
#pylint:disable=missing-class-docstring
#pylint:disable=missing-function-docstring
class CustomList(list):
    def first(self):
        # for example
        return self[0]
```
gives no message.
```
pylint 2.6.0
astroid 2.4.3
Python 3.8.2 (default, Sep 28 2020, 21:24:28) 
[GCC 8.3.0]
```

**Comment by pauladams8** [comment]
Sorry, should have been clearer. Upon further testing, the problem occurs only when I subclass the `List` alias in `typing` and with a type hint.

Try this:
```python
class TypeHint:
    pass

class CustomList(List[TypeHint]):
    def first(self):
        # pylint alerts me "Value 'self' is unsubscriptable", which is incorrect
        return self[0]
```

Additionally, pylint incorrectly reports additional errors when `List` is subclassed without a type hint.
- Inheriting 'List', which is not a class.
- Instance of '_GenericAlias' has no '{any member}' member when getting/setting custom properties

It seems these errors are not present with a type hint.

To reproduce this, use the sample above but without the `[TypeHint]`.

**Comment by pauladams8** [comment]
pylint also reports incorrectly on `List[TypeHint]` subclasses
> ` Value 'self' doesn't support membership test`

**Comment by hippo91** [comment]
@pauladams8 thanks for this update. I can now reproduce it.
While it is not linked to `python3.9` maybe the current work on supporting `typing` module under `python3.9` (#3890 #3905) may improve the situation.

**Comment by cdce8p** [comment]
@pauladams8 I wasn't able to reproduce any of the errors with the latest version of pylint. We improved the handling of the `typing` module quite a bit in recent versions so I would assume this has been fixed. Going to close this issue.

If you still experience issues, please report back and I'll reopen it.
